docs/bigquery-usage.rst

```diff
@@ -48,17 +48,58 @@ Authorization / Configuration
      >>> from gcloud import bigquery
      >>> client = bigquery.Client.from_service_account_p12('/path/to/creds.p12', 'jrandom@example.com')
 
-- Override the project inferred from the environment by passing an explicit
-  ``project`` to the constructor, or to either of the alternative classmethods:
+
+Projects
+--------
+
+A project is the top-level container in the ``BigQuery`` API:  it is tied
+closely to billing, and can provide default access control across all its
+datasets.  If no ``project`` is passed to the client container, the library
+attempts to infer a project using the environment (including explicit
+environment variables, GAE, and GCE).
+
+To override the project inferred from the environment, pass an explicit
+``project`` to the constructor, or to either of the alternative classmethods:
 
   .. doctest::
 
      >>> from gcloud import bigquery
      >>> client = bigquery.Client(project='PROJECT_ID')
 
+If no project is known, and it is not desired to infer one from the
+environment, pass an explicit value of ``None`` to the constructor.  Such a
+client can only be used to query for the list of projects to which the
+client's credentials has some access:
+
+  .. doctest::
+
+     >>> from gcloud import bigquery
+     >>> client = bigquery.Client(project=None)
+     >>> projects, next_page_token = client.list_projects()  # API request
```

Comment:
Same as with pubsub / storage:  either the cilent will have a successfully-inferred project (in which case the `list_projects()` call will succeed) or the `Client()` constructor will raise an error.  I'm suggesting that an explicit `project=None` should be the signal that the user wants to suppress the inference (which will require that the default value is actually a marker of some sort).